### PR TITLE
cmd/snap: support redirecting the client for testing

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

**Comment by chipaca** [comment]
:+1:

**Comment by chipaca** [comment]
:+1:

File: cmd/snap/main.go [hunk 1]
```diff
@@ -36,6 +37,14 @@ var optionsData options
 
 var parser = flags.NewParser(&optionsData, flags.HelpFlag|flags.PassDoubleDash)
 
+// ClientConfig is the configuration of the client.Client used by all commands.
```
Comment (pedronis) [inline cmd/snap/main.go]: just "for the Client" instead "of the client.Client", we see the full type below anyway

**Comment by pedronis** [comment]
nitpick, otherwise +1

**Pull request merged**